query/src/api/rpc/exchange/exchange_manager.rs

```diff
@@ -518,12 +530,17 @@ impl QueryCoordinator {
 
         let async_runtime = self.ctx.get_storage_runtime();
         let query_need_abort = self.ctx.query_need_abort();
+
         let executor =
             PipelineCompleteExecutor::from_pipelines(async_runtime, query_need_abort, pipelines)?;
         self.executor = Some(executor.clone());
 
-        Thread::named_spawn(Some(String::from("Distributed-Executor")), move || {
-            executor.execute().ok();
+        let exchange_manager = self.ctx.get_exchange_manager();
+        let query_id = self.query_id.clone();
+        Thread::named_spawn(Some("Distributed-Executor".to_string()), move || {
+            let result = executor.execute();
+
+            exchange_manager.on_finished_query(query_id.as_str(), &result.err());
```

Comment:
It's called in pipeline on finish in executor.

Reply:
Fixed.